File: src/main/java/org/digma/intellij/plugin/jaegerui/model/incoming/SpansMessage.java
package org.digma.intellij.plugin.jaegerui.model.incoming;

import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;


@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record SpansMessage(@JsonProperty(value = "action",required = true) String action,
                           @JsonProperty(value = "payload",required = true) SpanListPayload payload) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public SpansMessage {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/SpansMessage.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.*
import java.beans.ConstructorProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class SpansMessage
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("action", "payload")
constructor(
    val action: String,
    val payload: SpanListPayload
)